### ISSUE TITLE
A requst for details about changes in version 1.5

### ISSUE THREAD

**Issue by harrysarson** [karma-runner/karma-browserstack-launcher#144]
Hi, thanks for your work on this package!

The recently published changes (v1.5) cause the browerstack tests for the mathjs package to fail (for details see https://github.com/josdejong/mathjs/pull/1453 and https://travis-ci.org/josdejong/mathjs/jobs/511546544). I checked both the changlog and releases for this package and there is no mention there about the changes introduced between v1.4 and v1.5.

Would it be possible to share the changes with us to help us to understand why our tests fail with the latest release?

---

A sample of the errors we are getting:
``` 
26 03 2019 15:26:08.417:ERROR [launcher]: Cannot load browser "bs_firefox_android"!
  Error: Username is required.
    at ApiClient.BaseClient (/home/travis/build/josdejong/mathjs/node_modules/browserstack/lib/client.js:18:9)
    at ApiClient.ApiBaseClient (/home/travis/build/josdejong/mathjs/node_modules/browserstack/lib/api.js:9:13)
    at new ApiClient (/home/travis/build/josdejong/mathjs/node_modules/browserstack/lib/api.js:162:17)
    at Object.createClient (/home/travis/build/josdejong/mathjs/node_modules/browserstack/lib/api.js:278:10)
    at createBrowserStackClient (/home/travis/build/josdejong/mathjs/node_modules/karma-browserstack-launcher/index.js:64:20)
    at Array.invoke (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:75:15)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:48:43)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:54:19)
    at get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:54:19)
    at /home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:71:14
    at Array.map (<anonymous>)
    at invoke (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:70:31)
    at Array.instantiate (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:59:20)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:48:43)
    at module.(anonymous function) (/home/travis/build/josdejong/mathjs/node_modules/karma/lib/config.js:249:64)
    at Array.invoke (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:75:15)
26 03 2019 15:26:08.418:ERROR [launcher]: Cannot load browser "bs_firefox_windows"!
  Error: Can not resolve circular dependency! (Resolving: browserStackClient -> browserStackClient)
    at error (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:22:12)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:44:15)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:54:19)
    at get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:54:19)
    at /home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:71:14
    at Array.map (<anonymous>)
    at invoke (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:70:31)
    at Array.instantiate (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:59:20)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:48:43)
    at module.(anonymous function) (/home/travis/build/josdejong/mathjs/node_modules/karma/lib/config.js:249:64)
    at Array.invoke (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:75:15)
    at Injector.get (/home/travis/build/josdejong/mathjs/node_modules/di/lib/injector.js:48:43)
    at /home/travis/build/josdejong/mathjs/node_modules/karma/lib/launcher.js:62:72
    at names.forEach (/home/travis/build/josdejong/mathjs/node_modules/karma/lib/launcher.js:102:71)
    at Array.forEach (<anonymous>)
    at Launcher.launch (/home/travis/build/josdejong/mathjs/node_modules/karma/lib/launcher.js:102:13)
```

**Comment by Pringels** [comment]
+1 

The update to 1.5 also causes our tests to fail, as Browserstack can no longer connect to the Karma server on our CI agents.

**Comment by papandreou** [comment]
Same here: https://travis-ci.org/unexpectedjs/unexpected/jobs/512103840#L538

**Comment by nick-woodward** [comment]
Hey everyone! 

It looks like @Varun-garg may have unintentionally introduced a breaking change in the following commit (https://github.com/karma-runner/karma-browserstack-launcher/commit/ea4ec78352d8829742ae3e96b42944f96d0174e8).

Specifically he changed the environment variables for username and access key.

`BROWSER_STACK_USERNAME` became `BROWSERSTACK_USERNAME`
`BROWSER_STACK_ACCESS_KEY` became `BROWSERSTACK_ACCESS_KEY`

**Comment by nick-woodward** [comment]
Given @Varun-garg works for BrowserStack and based on the documentation for the library he switched to [browserstack-local](https://www.npmjs.com/package/browserstack-local) (owned by BrowserStack).

I'm guessing the change was intentional. 
That library by default looks for the following environment variables:

- `BROWSERSTACK_USERNAME`
- `BROWSERSTACK_ACCESS_KEY`

So probably to stay consistent across the board he changed it.
Granted it should have been a major version bump to this library.

**Comment by Varun-garg** [comment]
Thanks for looping me in @nick-woodward . Removing existing environment variables was unintentional, will check and roll out a fix for this soon.

**Comment by Varun-garg** [comment]
Regarding details of the official package, we were totally fine using the other one too. We started getting some customer complaints with it crashing sometimes downloading the local binaries, and after switching some customers also noticed slight performance improvements which may be because the official package is used across other JS frameworks and examples and is frequently updated addressing common issues.

**Comment by kalamcs** [comment]
This is now resolved in 1.5.1

**Comment by harrysarson** [comment]
Thanks! Our CI is now green for version 1.5.1.

@Varun-garg and @kalamcs if I may have one last request, would it be possible to update the changelog which is still at v1.4.0?
